prime-router/docs/proposals/0020-univeral-pipeline-topics.md

```diff
@@ -0,0 +1,63 @@
+  datatypes as they set up their system. Therefore, the information that tells us where certain data goes should be tied
+  to senders and receivers, and may not be solely determinable from the data itself. Topics should be an abstraction.
+
+## Proposal
+
+Currently, the Universal Pipeline (UP) is used for both use cases, and organization settings specify to use the UP by
+setting the `topic` field to `full-elr`. This proposal suggests retaining the `full-elr` topic, but also adding a `etor`
+topic. Those topics will both signify usage of the UP, however senders for one topic will only be able to send data to
+receivers of the same topic. Additionally, each topic may have a different set of default receivers; for instance, the
+current default quality filter for the `full-elr` topic expects a patient to have contact information, but for newborn
+screening ETOR requests, that information won't be available. So a separate default quality filter for the `etor` topic
+could set a reasonable starting set of quality expectations that are reasonable for ETOR requests.
+
+### Implications on internal types
+
+Currently, senders in the code are represented by a few different types. The overall parent class is called `Sender`.
+`Sender` is an abstract class which is extended by `FullELRSender` (used for UP) and `TopicSender` (used for the legacy
+pipeline). `TopicSender` has subtypes `CovidSender` and `MonkeypoxSender`.
+
+Under this proposal, `FullELRSender` will be renamed to `UniversalPipelineSender`, and `TopicSender` will be renamed
+to `LegacyPipelineSender` (still with subclasses `CovidSender` and `MonkeypoxSender`). Topic will still be used to
+determine which subclass of `Sender` to use; `full-elr` and `etor` will correspond to `UniversalPipelineSender`,
+while `covid-19` will correspond to `CovidSender` and `monkeypox` will correspond to `MonkeypoxSender`.
+
+Receivers don't have the same kind of structure as senders, so the `Receiver` class and subclasses will not have the
+same kinds of changes. Instead, during routing in the UP, the topic of the receiver will be considered so that reports
+from senders only go to receivers with matching topics. This evaluation should occur prior to the evaluation of any
+filters for performance reasons.
+
+### Implications on organization settings
+
+After the changes here, the existing settings for ETOR senders and receivers will largely still hold, with the exception
+of the value of `topic`, which will be `etor` instead of `full-elr`.
+
+Since ELR senders and receivers have a topic of `full-elr` already, they will not require any changes.
+
+Monkeypox and Covid-19 are special cases initially, but eventually, those will be phased out, and condition-specific
+filtering will only occur in the Universal Pipeline by using condition filters. However, to minimize impact in the
+short-term, we will continue to support `covid-19` and `monkeypox` as topics, so those settings will not require
```

Comment:
Might be fine, but wanted to bring up that if covid and monkeypox end up being moved to UP down the road, we will need to either use a transitionary topic name or change its implementation for everybody at once.

Reply:
Yea we would probably want some kind of rollout plan based on what the current state of senders/receivers is, but especially with the covid emergency wrapped up, it might not be too disruptive to just move these topics to the UP all at once.